### ISSUE TITLE
ERROR: Failed building wheel for pylzma

### ISSUE THREAD

**Issue by veryspecialagent77** [bkerler/edl#181]
Setting this up on an Asus running Win 10. I get this error during "pip3 install -r requirements.txt"

Requirement already satisfied: wheel in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 1)) (0.37.0)
Requirement already satisfied: pyusb>=1.1.0 in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 2)) (1.2.1)
Requirement already satisfied: pyserial>=3.4 in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 3)) (3.5)
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2-py2.py3-none-any.whl
Collecting pylzma>=0.5.0
  Using cached pylzma-0.5.0.tar.gz (4.2 MB)
  Preparing metadata (setup.py) ... done
Collecting pycryptodome
  Using cached pycryptodome-3.11.0-cp35-abi3-win_amd64.whl (1.8 MB)
Collecting lxml>=4.6.1
  Using cached lxml-4.6.4-cp38-cp38-win_amd64.whl (3.6 MB)
Collecting colorama
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Requirement already satisfied: usb in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 9)) (0.0.83.dev0)
Collecting capstone
  Using cached capstone-4.0.2-py2.py3-none-win_amd64.whl (896 kB)
Collecting keystone-engine
  Using cached keystone_engine-0.9.2-py2.py3-none-win_amd64.whl (1.4 MB)
Building wheels for collected packages: pylzma
  Building wheel for pylzma (setup.py) ... error
  **ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ryans\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"'; __file__='"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\ryans\AppData\Local\Temp\pip-wheel-epzt_gea'
       cwd: C:\Users\ryans\AppData\Local\Temp\pip-install-r5y7uysw\pylzma_397065a4f6184ebe9cdf56e9b575b197\
  Complete output (20 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.8
  copying py7zlib.py -> build\lib.win-amd64-3.8
  running build_ext
  adding support for multithreaded compression
  building 'pylzma' extension
  creating build\temp.win-amd64-3.8
  creating build\temp.win-amd64-3.8\Release
  creating build\temp.win-amd64-3.8\Release\src
  creating build\temp.win-amd64-3.8\Release\src\pylzma
  creating build\temp.win-amd64-3.8\Release\src\sdk
  creating build\temp.win-amd64-3.8\Release\src\sdk\C
  creating build\temp.win-amd64-3.8\Release\src\compat
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -DCOMPRESS_MF_MT=1 -Isrc/sdk/C "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsrc/pylzma/pylzma.c /Fobuild\temp.win-amd64-3.8\Release\src/pylzma/pylzma.obj
  pylzma.c
  C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for pylzma**
  Running setup.py clean for pylzma
Failed to build pylzma
Installing collected packages: pylzma, pycryptodome, lxml, keystone-engine, docopt, colorama, capstone
    Running setup.py install for pylzma ... error
    **ERROR: Command errored out with exit status 1:
     command: 'C:\Users\ryans\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"'; __file__='"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\ryans\AppData\Local\Temp\pip-record-lye0yrjh\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\ryans\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Include\pylzma'
         cwd: C:\Users\ryans\AppData\Local\Temp\pip-install-r5y7uysw\pylzma_397065a4f6184ebe9cdf56e9b575b197\
    Complete output (20 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.8
    copying py7zlib.py -> build\lib.win-amd64-3.8
    running build_ext
    adding support for multithreaded compression
    building 'pylzma' extension
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\src
    creating build\temp.win-amd64-3.8\Release\src\pylzma
    creating build\temp.win-amd64-3.8\Release\src\sdk
    creating build\temp.win-amd64-3.8\Release\src\sdk\C
    creating build\temp.win-amd64-3.8\Release\src\compat
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -DCOMPRESS_MF_MT=1 -Isrc/sdk/C "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsrc/pylzma/pylzma.c /Fobuild\temp.win-amd64-3.8\Release\src/pylzma/pylzma.obj
    pylzma.c
    C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\ryans\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"'; __file__='"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\ryans\AppData\Local\Temp\pip-record-lye0yrjh\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\ryans\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Include\pylzma' Check the logs for full command output.**

**Comment by veryspecialagent77** [comment]
Setting this up on an Asus GL753 running Win 10. I get this error during "`pip3 install -r requirements.txt`"

Requirement already satisfied: wheel in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 1)) (0.37.0)
Requirement already satisfied: pyusb>=1.1.0 in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 2)) (1.2.1)
Requirement already satisfied: pyserial>=3.4 in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 3)) (3.5)
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2-py2.py3-none-any.whl
Collecting pylzma>=0.5.0
  Using cached pylzma-0.5.0.tar.gz (4.2 MB)
  Preparing metadata (setup.py) ... done
Collecting pycryptodome
  Using cached pycryptodome-3.11.0-cp35-abi3-win_amd64.whl (1.8 MB)
Collecting lxml>=4.6.1
  Using cached lxml-4.6.4-cp38-cp38-win_amd64.whl (3.6 MB)
Collecting colorama
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Requirement already satisfied: usb in c:\users\ryans\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 9)) (0.0.83.dev0)
Collecting capstone
  Using cached capstone-4.0.2-py2.py3-none-win_amd64.whl (896 kB)
Collecting keystone-engine
  Using cached keystone_engine-0.9.2-py2.py3-none-win_amd64.whl (1.4 MB)
Building wheels for collected packages: pylzma
  Building wheel for pylzma (setup.py) ... error
  **ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ryans\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"'; __file__='"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\ryans\AppData\Local\Temp\pip-wheel-epzt_gea'
       cwd: C:\Users\ryans\AppData\Local\Temp\pip-install-r5y7uysw\pylzma_397065a4f6184ebe9cdf56e9b575b197\
  Complete output (20 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.8
  copying py7zlib.py -> build\lib.win-amd64-3.8
  running build_ext
  adding support for multithreaded compression
  building 'pylzma' extension
  creating build\temp.win-amd64-3.8
  creating build\temp.win-amd64-3.8\Release
  creating build\temp.win-amd64-3.8\Release\src
  creating build\temp.win-amd64-3.8\Release\src\pylzma
  creating build\temp.win-amd64-3.8\Release\src\sdk
  creating build\temp.win-amd64-3.8\Release\src\sdk\C
  creating build\temp.win-amd64-3.8\Release\src\compat
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -DCOMPRESS_MF_MT=1 -Isrc/sdk/C "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsrc/pylzma/pylzma.c /Fobuild\temp.win-amd64-3.8\Release\src/pylzma/pylzma.obj
  pylzma.c
  C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for pylzma**
  Running setup.py clean for pylzma
Failed to build pylzma
Installing collected packages: pylzma, pycryptodome, lxml, keystone-engine, docopt, colorama, capstone
    Running setup.py install for pylzma ... error
    **ERROR: Command errored out with exit status 1:
     command: 'C:\Users\ryans\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"'; __file__='"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\ryans\AppData\Local\Temp\pip-record-lye0yrjh\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\ryans\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Include\pylzma'
         cwd: C:\Users\ryans\AppData\Local\Temp\pip-install-r5y7uysw\pylzma_397065a4f6184ebe9cdf56e9b575b197\
    Complete output (20 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.8
    copying py7zlib.py -> build\lib.win-amd64-3.8
    running build_ext
    adding support for multithreaded compression
    building 'pylzma' extension
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\src
    creating build\temp.win-amd64-3.8\Release\src\pylzma
    creating build\temp.win-amd64-3.8\Release\src\sdk
    creating build\temp.win-amd64-3.8\Release\src\sdk\C
    creating build\temp.win-amd64-3.8\Release\src\compat
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -DCOMPRESS_MF_MT=1 -Isrc/sdk/C "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsrc/pylzma/pylzma.c /Fobuild\temp.win-amd64-3.8\Release\src/pylzma/pylzma.obj
    pylzma.c
    C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\ryans\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"'; __file__='"'"'C:\\Users\\ryans\\AppData\\Local\\Temp\\pip-install-r5y7uysw\\pylzma_397065a4f6184ebe9cdf56e9b575b197\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\ryans\AppData\Local\Temp\pip-record-lye0yrjh\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\ryans\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Include\pylzma' Check the logs for full command output.**

**Comment by veryspecialagent77** [comment]
I apologize for the formatting. I had intended for just the errors (in red on my screen) to be bold and it all be one font and size. Now it's 5am and time to take a short nap before work.

**Comment by bkerler** [comment]
As the readme says. Don't use python from the windows store as it's utterly broken. Use python from python.org.

**Comment by veryspecialagent77** [comment]
Change the instructions to say "Do not install from windows store because of won't work" not "but that step isn't required." The way it is written makes it seem like you can and that the error is expected.

As you can see from the bottom log, I uninstalled and reinstalled everything per the directions. Still got the same errors. 

I pulled up and am going through the logs. For starters, I didn't have everything I needed downloaded from Visual Studio. I'll update this with the fix when I figure it out. I only pasted here in the event someone else might have already dealt with it and hadn't posted about it.

**Comment by veryspecialagent77** [comment]
Figured it out, in case anyone else has this issue: 
Windows SDK hadn't installed in VS2019. Installed it, STILL got similar error. Main issue seemed to be it couldn't find requrements.txt
requirements.txt was in Modem folder, but only had one line. Couldn't be it.
I found the requirements.txt I needed (that had the missing info stated in the error message) stashed away in a temp folder created during installation. Copied it to the edl folder, renamed it requirements.txt, ran the pip3 install -r requirements.txt and was great success.
